Gems/EMotionFX/Code/EMotionFX/Pipeline/RCExt/Actor/ActorBuilder.cpp

```diff
@@ -206,37 +206,42 @@ namespace EMotionFX
             if (skeletonOptimizationRule)
             {
                 const SceneData::SceneNodeSelectionList& criticalBonesList = skeletonOptimizationRule->GetCriticalBonesList();
-                const size_t criticalBoneCount = criticalBonesList.GetSelectedNodeCount();
-                
-                for (size_t boneIndex = 0; boneIndex < criticalBoneCount; ++boneIndex)
-                {
-                    const AZStd::string& bonePath = criticalBonesList.GetSelectedNode(boneIndex);
 
-                    // The bone name stores the node path in the scene. We need to convert it to EMotionFX node name.
-                    const SceneContainers::SceneGraph::NodeIndex nodeIndex = graph.Find(bonePath);
-                    if (!nodeIndex.IsValid())
+                criticalBonesList.EnumerateSelectedNodes(
+                    [&](const AZStd::string& bonePath)
                     {
-                        AZ_TracePrintf(SceneUtil::WarningWindow, "Critical bone %s is not stored in the scene. Skipping it.", bonePath.c_str());
-                        continue;
-                    }
+                        // The bone name stores the node path in the scene. We need to convert it to EMotionFX node name.
+                        const SceneContainers::SceneGraph::NodeIndex nodeIndex = graph.Find(bonePath);
+                        if (!nodeIndex.IsValid())
+                        {
+                            AZ_TracePrintf(
```

Comment:
`AZ_TracePrintf` is deprecated, use `AZ_Trace`
```suggestion
                            AZ_Trace(
```

Reply:
Yeah, none of these were introduced in this PR, I just took existing code and moved it into Enumerate* methods. I wasn't trying to clean up the existing code. But since you've provided these fixups so nicely, I'll consume them into the PR. :)